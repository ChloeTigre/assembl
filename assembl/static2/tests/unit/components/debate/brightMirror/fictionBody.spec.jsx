// @flow
import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import PostBody from '../../../../../js/app/components/debate/common/post/postBody';
import FictionBody from '../../../../../js/app/components/debate/brightMirror/fictionBody';
import type { FictionBodyProps } from '../../../../../js/app/components/debate/brightMirror/fictionBody';

configure({ adapter: new Adapter() });

const defaultId: string = '0';
const defaultTitle: string = 'Fugit veritatis nemo';
const defaultContent: string = 'Vero et ut et quia quo. Molestiae ut cupiditate odio numquam veniam esse cumque modi.';
const defaultLocale: string = 'en';
const defaultExtracts: Array<FictionExtractFragment> = [];
const defaultIsAuthorAccountDeleted: boolean = false;
const defaultDbId: number = 1;
const bodyMimeType: string = 'text/html';
const refetchPost: Function = jest.fn();

const defaultFictionBody: FictionBodyProps = {
  id: defaultId,
  title: defaultTitle,
  content: defaultContent,
  contentLocale: defaultLocale,
  lang: defaultLocale,
  extracts: defaultExtracts,
  isAuthorAccountDeleted: defaultIsAuthorAccountDeleted,
  dbId: defaultDbId,
  bodyMimeType: bodyMimeType,
  refetchPost: refetchPost
};

describe('<FictionBody /> - with shallow', () => {
  let wrapper;
  let fictionBody: FictionBodyProps;

  beforeEach(() => {
    window.getSelection = () => ({
      removeAllRanges: () => {}
    });
    // disableLifecycleMethods to prevent consulting the DOM
    wrapper = shallow(<FictionBody {...fictionBody} />, { disableLifecycleMethods: true });
  });

  it('should render post body', () => {
    expect(wrapper.find(PostBody)).toHaveLength(1);
  });
});

describe('<FictionBody /> - with mount', () => {
  let wrapper;
  let fictionBody: FictionBodyProps;

  beforeEach(() => {
    window.getSelection = () => ({
      removeAllRanges: () => {}
    });
    // Create DOM to allow document.getElementById function
    const div = document.createElement('div');
    window.domNode = div;
    // $FlowFixMe because document.body may be null
    document.body.appendChild(div);
    fictionBody = { ...defaultFictionBody };
    wrapper = mount(<FictionBody {...fictionBody} />, { attachTo: window.domNode });
  });

  it('should display the fiction content', () => {
    const fictionContent: string = wrapper.find('div [className="post-body-content body"]').text();
    expect(fictionContent).toEqual(defaultContent);
  });

  it('should display "no content specified" when content is set to null', () => {
    wrapper.setProps({ content: '' });
    const fictionContent: string = wrapper.find('div [className="post-body-content body"]').text();
    expect(fictionContent).toEqual('no content specified');
  });
});

describe('<FictionBody /> - snapshots', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<FictionBody {...defaultFictionBody} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot no title nor content', () => {
    const props: FictionBodyProps = {
      title: '',
      content: '',
      id: defaultId,
      contentLocale: defaultLocale,
      lang: defaultLocale,
      extracts: defaultExtracts,
      isAuthorAccountDeleted: defaultIsAuthorAccountDeleted,
      dbId: defaultDbId,
      bodyMimeType: bodyMimeType,
      refetchPost: refetchPost
    };
    const component = renderer.create(<FictionBody {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});