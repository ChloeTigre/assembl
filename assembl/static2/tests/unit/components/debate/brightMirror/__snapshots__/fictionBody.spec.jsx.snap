// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FictionBody /> - snapshots should match snapshot 1`] = `
<div>
  <div>
    <div
      className="side-comment-container"
    />
  </div>
  <div
    className="post-body post-body--is-harvestable"
  >
    <h3
      className="post-body-title dark-title-3"
    >
      Fugit veritatis nemo
    </h3>
    <div
      className="post-body-content body"
    >
      <div
        onMouseUp={[Function]}
      >
        <div
          id="message-body-local:Content/1"
        >
          Vero et ut et quia quo. Molestiae ut cupiditate odio numquam veniam esse cumque modi.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FictionBody /> - snapshots should match snapshot no title nor content 1`] = `
<div>
  <div>
    <div
      className="side-comment-container"
    />
  </div>
  <div
    className="post-body post-body--is-harvestable"
  >
    <h3
      className="post-body-title dark-title-3"
    >
      no title specified
    </h3>
    <div
      className="post-body-content body"
    >
      <div
        onMouseUp={[Function]}
      >
        <div
          id="message-body-local:Content/1"
        >
          no content specified
        </div>
      </div>
    </div>
  </div>
</div>
`;
